org.osgi.test.common/src/main/java/org/osgi/test/common/annotation/config/ConfigEntry.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) OSGi Alliance (2020). All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.osgi.test.common.annotation.config;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+@Inherited
+@Target({
+	ANNOTATION_TYPE
+})
+@Retention(RUNTIME)
+@Documented
+public @interface ConfigEntry {
+
+	String key();
+
+	String value() default "";
+
+	String type() default "String";
```

Comment:
I think this should should be an inner enum. See https://docs.osgi.org/specification/osgi.core/7.0.0/framework.module.html#i3217016 for the allowable types in a configuration.

Reply:
See: https://github.com/osgi/osgi-test/pull/165/commits/4b57e835d598bdbdc72f515e916dbc74fd1f76be#diff-955bbd7d7884b63d979a2386d67dac99

https://github.com/osgi/osgi-test/pull/165/commits/4b57e835d598bdbdc72f515e916dbc74fd1f76be#diff-8abbc06a0596380453c03014d8e0ca12R454-R532